File: RepoTransBench/source_projects/Java/Netflix_dyno-queues/dyno-queues-core/src/test/java/com/netflix/dyno/queues/ShardSupplierPublicTest.java
package com.netflix.dyno.queues;

import org.junit.Test;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

import static org.junit.Assert.*;

public class ShardSupplierPublicTest {

    private static class TestShardSupplier implements ShardSupplier {
        private Set<String> shards;
        private String currentShard;
        public TestShardSupplier(Set<String> shards, String currentShard) {
            this.shards = shards;
            this.currentShard = currentShard;
        }
        @Override
        public Set<String> getQueueShards() { return shards; }
        @Override
        public String getCurrentShard() { return currentShard; }
        @Override
        public String getShardForHost(com.netflix.dyno.connectionpool.Host host) {
            // return host's hostname for testing public case
            return host.getHostName();
        }
    }

    @Test
    public void testShardSupplierInterface() {
        Set<String> shards = new HashSet<>();
        shards.add("alpha");
        shards.add("beta");
        TestShardSupplier supplier = new TestShardSupplier(shards, "beta");

        assertEquals(shards, supplier.getQueueShards());
        assertEquals("beta", supplier.getCurrentShard());
        com.netflix.dyno.connectionpool.Host host = new com.netflix.dyno.connectionpool.Host("hostY", 2, "rackX", com.netflix.dyno.connectionpool.Host.Status.Up);
        assertEquals("hostY", supplier.getShardForHost(host));
    }

    @Test
    public void testShardSupplierEmptySet() {
        TestShardSupplier supplier = new TestShardSupplier(Collections.emptySet(), null);
        assertTrue(supplier.getQueueShards().isEmpty());
        assertNull(supplier.getCurrentShard());
    }
}

File: RepoTransBench/target_projects/Java/Go/Netflix_dyno-queues/public_tests/shard_supplier_public_test.go
package public_tests

import (
	"testing"
)

type Host struct {
	hostname string
	port     int
	rack     string
	status   string
}

type ShardSupplier interface {
	GetQueueShards() map[string]struct{}
	GetCurrentShard() string
	GetShardForHost(host Host) string
}

type TestShardSupplier struct {
	shards       map[string]struct{}
	currentShard string
}

func NewTestShardSupplier(shards []string, currentShard string) *TestShardSupplier {
	set := make(map[string]struct{})
	for _, s := range shards {
		set[s] = struct{}{}
	}
	return &TestShardSupplier{shards: set, currentShard: currentShard}
}

func (t *TestShardSupplier) GetQueueShards() map[string]struct{} { return t.shards }
func (t *TestShardSupplier) GetCurrentShard() string             { return t.currentShard }
func (t *TestShardSupplier) GetShardForHost(host Host) string    { return host.hostname }

func TestShardSupplierPublic_Interface(t *testing.T) {
	shards := []string{"alpha", "beta"}
	supplier := NewTestShardSupplier(shards, "beta")
	gotShards := supplier.GetQueueShards()
	for _, k := range shards {
		if _, ok := gotShards[k]; !ok {
			t.Errorf("Shard %s not found", k)
		}
	}
	if supplier.GetCurrentShard() != "beta" {
		t.Errorf("Expected 'beta' as currentShard, got %s", supplier.GetCurrentShard())
	}
	host := Host{hostname: "hostY", port: 2, rack: "rackX", status: "Up"}
	if supplier.GetShardForHost(host) != "hostY" {
		t.Errorf("Expected host's hostname 'hostY', got %s", supplier.GetShardForHost(host))
	}
}

func TestShardSupplierPublic_EmptySet(t *testing.T) {
	supplier := NewTestShardSupplier([]string{}, "")
	if len(supplier.GetQueueShards()) != 0 {
		t.Errorf("Expected empty shards set")
	}
	if supplier.GetCurrentShard() != "" {
		t.Errorf("Expected nil current shard")
	}
}